internal/node/node.bzl

```diff
@@ -52,6 +54,15 @@ def _compute_node_modules_roots(ctx):
                 if other_workspace != workspace:
                     fail("All npm dependencies at the path '%s' must come from a single workspace. Found '%s' and '%s'." % (path, other_workspace, workspace))
             node_modules_roots[path] = workspace
+
+    # Add in roots for multi-linked first party deps
+    for dep in ctx.attr.data:
```

Comment:
I was deliberate so that first party deps roots are scanned 2nd and the logic depends on what 3rd party deps roots were added. I'll revisit this with the directory artifacts change upcoming and optimize at that point and 3p and 1p deps will be handled similarly with that chagne.